kernel: bump 6.12 to .52 .53 and .54

EPinci opened the pull request:
kernel: bump 6.12 to 6.12.52

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.12.52

All patches automatically rebased.

Build system: x86/64
Build-tested: mediatek/filogic
Run-tested: mediatek/filogic

---

kernel: bump 6.12 to 6.12.53
    
Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.12.53

Dropped patch
- /target/linux/generic/backport-6.12/541-v6.18-ksmbd-add-max-ip-connections-parameter.patch
merged upstream with commit https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-6.12.y&id=8173dcfafe116adb08f8daf21c09c71ac5882d8f

Added backport
- target/linux/generic/backport-6.12/510-v6.18-ksmbd-fix-recursive-locking-in-RPC-handle-list-access.patch
cherry picked from https://github.com/torvalds/linux/commit/88f170814fea74911ceab798a43cbd7c5599bed4

All other patches automatically rebased.
   
Build system: x86/64
Build-tested: mediatek/filogic
Run-tested: mediatek/filogic

---

kernel: bump 6.12 to 6.12.54

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.12.54

Manually rebased:
bcm27xx/patches-6.12/950-0018-arm64-setup-Fix-build-warning.patch

All other patches automatically rebased.

Build system: x86/64
Build-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, x86/64-glibc
Run-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, x86/64-glibc

Build system: x86/64
Build-tested: x86/64-glibc
Run-tested: x86/64-glibc

Build system: x86/64
Build-tested: mediatek/filogic
Run-tested: mediatek/filogic

EPinci committed:
kernel: bump 6.12 to 6.12.52

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.12.52

All patches automatically rebased.

Build system: x86/64
Build-tested: mediatek/filogic
Run-tested: mediatek/filogic

Signed-off-by: Edoardo Pinci <epinci@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/20408
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

graysky2 commented:
Yeah, I tried too but those two reverts are over my head.

... why are we reverting them?

pesa1234 commented:
> > Yeah, I tried too but those two reverts are over my head.
> 
> For .53, I pinned it down to two problematic patches that I don't know how to handle:
> 
>     * target/linux/generic/hack-6.12/940-Revert-ksmbd-extend-the-connection-limiting-mechanism-to-ipv6.patch
> 
>     * target/linux/generic/hack-6.12/941-Revert-ksmbd-limit-repeated-connections-from-clients.patch
> 
> 
> Beside those, I have 4 automatic refreshes. Shall I push here what I have so far? Who can be autoritative for those patches?

On 6.12.53 that two patches can be deleted, the fix is merged upstream with this [commit](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-6.12.y&id=8173dcfafe116adb08f8daf21c09c71ac5882d8f)

EPinci committed:
kernel: bump 6.12 to 6.12.53

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.12.53

Dropped patch
- /target/linux/generic/backport-6.12/541-v6.18-ksmbd-add-max-ip-connections-parameter.patch
merged upstream with commit https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-6.12.y&id=8173dcfafe116adb08f8daf21c09c71ac5882d8f

Added backport
- target/linux/generic/backport-6.12/510-v6.18-ksmbd-fix-recursive-locking-in-RPC-handle-list-access.patch
cherry picked from https://github.com/torvalds/linux/commit/88f170814fea74911ceab798a43cbd7c5599bed4

All other patches automatically rebased.

Build system: x86/64
Build-tested: mediatek/filogic
Run-tested: mediatek/filogic

Signed-off-by: Edoardo Pinci <epinci@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/20408
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

EPinci commented:
@pesa1234 After your last commit to main
`/target/linux/generic/backport-6.12/541-v6.18-ksmbd-add-max-ip-connections-parameter.patch`
is now breaking up with .53. Can you advise?

EPinci commented:
@pesa1234 Looks like that new patch is already merged upstream in .53 with the same commit you referenced before, I'm dropping it. Can you, please, confirm?

pesa1234 commented:
> @pesa1234 Looks like that new patch is already merged upstream in .53 with the same commit you referenced before, I'm dropping it. Can you, please, confirm?

Only this https://github.com/openwrt/openwrt/blob/e78f000869f1858b9bdf2a72daf2c7e27bfb4b82/target/linux/generic/backport-6.12/541-v6.18-ksmbd-add-max-ip-connections-parameter.patch

Borromini commented:
Compile-tested: mvebu (RB5009UG+S+IN), rtl930x (XGS1210-12 B1)
Run-tested: mvebu (RB5009UG+S+IN), rtl930x (XGS1210-12 B1)

hardfalcon commented:
Compile-tested: x86/64-musl, realtek/rtl838x, ath79/generic
Run-tested: x86/64-musl (qemu microvm), realtek/rtl838x (Netgear GS108T v3), ath79/generic (TP-Link Archer C7 v4)

graysky2 committed:
kernel: bump 6.12 to 6.12.54

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.12.54

Manually rebased:
	bcm27xx/patches-6.12/950-0018-arm64-setup-Fix-build-warning.patch

All other patches automatically rebased.

Build system: x86/64
Build-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, x86/64-glibc
Run-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, x86/64-glibc

Build system: x86/64
Build-tested: x86/64-glibc
Run-tested: x86/64-glibc

Signed-off-by: John Audia <therealgraysky@proton.me>

Build system: x86/64
Build-tested: mediatek/filogic
Run-tested: mediatek/filogic

Signed-off-by: Edoardo Pinci <epinci@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/20408
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

hauke commented:
Pleas run `make target/{clean,refresh}`.

EPinci commented:
I think something is not behaving right with the runners?
Queued tasks are OK but prechecks are stuck in "starting" for hours until they timeout.
<img width="954" height="376" alt="image" src="https://github.com/user-attachments/assets/d6e434ed-8a2d-4af4-a251-740b1a8b6e7a" />

DragonBluep commented on target/linux/generic/backport-6.12/510-v6.18-ksmbd-fix-recursive-locking-in-RPC-handle-list-access.patch:
```diff
@@ -0,0 +1,155 @@
+
+diff --git a/fs/smb/server/mgmt/user_session.c b/fs/smb/server/mgmt/user_session.c
+index 6fa025374f2f3a..1c181ef9992957 100644
+--- a/fs/smb/server/mgmt/user_session.c
```
Please remove .54 kernel update commit and run `make target/linux/{clean,refresh}`. The we can apply .54 kernel update again.

EPinci replied:
Ok, I undone .54 and run the refresh. Found that missed patch and amended my local repo.
Do you want me to force push first without .54 or shall I go ahead, reapply it and repush?

DragonBluep replied:
Push the final .54 version please.

schuettecarsten commented:
Compile and ruin tested on bcm27xx/bcm2711.

GoetzGoerisch reviewed (approved):
run-tested: HPE 1920-24G (JG924A)
